python/lsst/daf/butler/script/retrieveArtifacts.py

```diff
@@ -114,8 +118,15 @@ def retrieveArtifacts(
     )
     refs = list(itertools.chain(*query.getDatasets()))
     log.info("Number of datasets matching query: %d", len(refs))
+    if not refs:
+        return []
+
+    if not zip:
+        transferred = butler.retrieveArtifacts(
+            refs, destination=destination, transfer=transfer, preserve_path=preserve_path, overwrite=clobber
+        )
+    else:
+        zip_file = butler.retrieve_artifacts_zip(refs, destination=destination)
```

Comment:
Yes. Maybe I should implement clobber param.

Reply:
Implemented, although in an ideal world it would complain before it spent a long time downloading all the files. Currently it downloads all the files and then complains because the zip file is constructed from the files within the zip.